hoomd/md/compute.py

```diff
@@ -282,84 +283,71 @@ def num_particles(self):
+    displacements from the lattice positions, which are saved when the
+    :py:class:`ThermoHMA` is instantiated. :py:class:`ThermoHMA` objects can be
+    added to a logger for logging during a simulation, see
+    :py:class:`hoomd.logging.Logger` for more details.
 
     Examples::
 
-        g = group.all()
-        compute.thermoHMA(group=g, temperature=1.0)
+        hma = hoomd.compute.thermoHMA(filter=hoomd.filter.Type('A'), temperature=1.0)
     """
 
-    def __init__(self, group, temperature, harmonicPressure=0):
+    def __init__(self, filter, temperature, harmonic_pressure=0):
 
-        # initialize base class
-        _compute.__init__(self);
+        # store metadata
+        param_dict = ParameterDict(
+            temperature=float(temperature),
+            harmonic_pressure=float(harmonic_pressure)
+        )
+        # set defaults
+        self._param_dict.update(param_dict)
 
-        suffix = '';
-        if group.name != 'all':
-            suffix = '_' + group.name;
+        self._filter = filter
+        # initialize base class
+        super().__init__()
 
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_compute = _hoomd.ComputeThermoHMA(hoomd.context.current.system_definition, group.cpp_group, temperature, harmonicPressure, suffix);
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermoHMA_cls = _md.ComputeThermoHMA
         else:
-            self.cpp_compute = _hoomd.ComputeThermoHMAGPU(hoomd.context.current.system_definition, group.cpp_group, temperature, harmonicPressure, suffix);
-
-        hoomd.context.current.system.addCompute(self.cpp_compute, self.compute_name);
-
-        # save the group for later referencing
-        self.group = group;
-
-    def disable(self):
-        R""" Disables the thermoHMA.
-
-        Examples::
-
-            my_thermo.disable()
-
-        Executing the disable command will remove the thermoHMA compute from the system. Any ```hoomd.run``` command
-        executed after disabling a thermoHMA compute will not be able to log computed values with ``hoomd.analyze.log``.
-
-        A disabled thermoHMA compute can be re-enabled with :py:meth:`enable()`.
-        """
-
-        _compute.disable(self)
-
-    def enable(self):
-        R""" Enables the thermoHMA compute.
-
-        Examples::
+            thermoHMA_cls = _md.ComputeThermoHMAGPU
+        group = self._simulation.state._get_group(self._filter)
+        self._cpp_obj = thermoHMA_cls(self._simulation.state._cpp_sys_def,
+                                      group,
+                                      self.temperature,
+                                      self.harmonic_pressure,
+                                      "")
+        super()._attach()
 
-            my_thermo.enable()
+    @log
+    def potential_energyHMA(self):
```

Comment:
Since the namespace will include the class name, I don't think we need the `HMA` suffix